colossus/src/main/scala/colossus/service/ServiceClient.scala

```diff
@@ -216,30 +214,30 @@ extends Controller[O,I](codec, ControllerConfig(config.pendingBufferSize, config
     connectionAttempts = 0
   }
 
-  private def purgeBuffers(pendingException : => Throwable) {
+  private def purgeBuffers(reason : Throwable) {
     sentBuffer.foreach{s => 
       errors.hit(tags = hpTags)
```

Comment:
Maybe, I'm not sure if we should use the `DisconnectCause` that caused the purge (which as of now isn't passed into the function) or just the error itself.  Needs some more thought so I'll hold off on it for how.